machine/v1/types_controlplanemachineset.go

```diff
@@ -340,6 +340,15 @@ type RootVolume struct {
 	// +kubebuilder:validation:Pattern=`^[^ ]*$`
 	// +optional
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
+
+	// volumeType specifies the type of the root volume that will be provisioned.
+	// If not specifified, the root volume will be created as the type in the machine template.
+	// The maximum length of a volume type name is 255 as per the OpenStack limit.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=255
+	// +kubebuilder:validation:Pattern=`^[^ ]*$`
+	// +optional
+	VolumeType string `json:"volumeType,omitempty"`
```

Comment:
This corresponds exactly with our machinev1alpha1 API:
https://github.com/openshift/api/blob/master/machine/v1alpha1/types_openstack.go#L358